distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/platform/TestSecurity.java

```diff
@@ -1289,4 +1321,190 @@ public String sendNotPermittedRequest(String jolokiaEndpoint) {
                 .print();
     }
+
+    @Test
+    public void testAccessGroupsGuest() throws Exception {
+        getSecurityPolicy().configureWebContextPolicy("/=ldap,/solr=karaf,/services/csw=karaf",
+                null,
+                null,
+                createWhitelist("/services/secure,/services/public"));
+
+        final File definitionFile = ingestDefinitionJsonWithWaitCondition("injections.json", () -> {
+            expect("Injectable attributes to be registered").within(30, TimeUnit.SECONDS)
+                    .until(() -> getServiceManager().getServiceReferences(InjectableAttribute.class,
+                            null), hasSize(4));
+            return null;
+        });
+
+        List attr = ImmutableList.of("security.access-groups=accessGroup");
+
+        configureAuthZRealm(attr, getAdminConfig());
+
+        try {
+            String testData = IOUtils.toString(IOUtils.toInputStream(getFileContent(
+                    XML_RECORD_RESOURCE_PATH + "/accessGroupTokenMetacard.xml")));
+            testData = testData.replace(ACCESS_GROUP_REPLACE_TOKEN, "A");
+
+            String id = CatalogTestCommons.ingest(testData, MediaType.TEXT_XML);
+
+            String url = SERVICE_ROOT.getUrl() + "/catalog/query?q=" + id + "&src=local";
+            configureRestForBasic(SDK_SOAP_CONTEXT);
+            waitForSecurityHandlers(url);
+            getSecurityPolicy().waitForBasicAuthReady(url);
+
+            configureMetacardAttributeSecurityFiltering(attr,
+                    ImmutableList.of(""),
+                    getAdminConfig());
+
+            String response = given().auth()
+                    .basic(A_USER, USER_PASSWORD)
+                    .when()
+                    .get(url)
+                    .then()
+                    .log()
+                    .all()
+                    .assertThat()
+                    .statusCode(equalTo(HttpStatus.SC_OK))
+                    .extract()
+                    .response()
+                    .print();
+
+            assertThat(response, containsString("Lady Liberty"));
+
+            response = given().auth()
+                    .basic(B_USER, USER_PASSWORD)
+                    .when()
+                    .get(url)
+                    .then()
+                    .log()
+                    .all()
+                    .assertThat()
+                    .statusCode(equalTo(HttpStatus.SC_OK))
+                    .extract()
+                    .response()
+                    .print();
+
+            assertThat(response, (containsString("Lady Liberty")));
+
+        } finally {
+            getSecurityPolicy().configureWebContextPolicy("/=ldap,/solr=karaf",
+                    null,
+                    null,
+                    createWhitelist("/services/secure,/services/public"));
+            uninstallDefinitionJson(definitionFile, () -> {
+                expect("Injectable attributes to be unregistered").within(10, TimeUnit.SECONDS)
```

Comment:
Could create a helper method for this specific attribute unregistered check. Looks like you use it in quite a few places.